hs/SvgParsing/ParserUtil.hs

```diff
@@ -74,27 +74,27 @@ getChildren = path [children]
 
 -- | Gets the value of the attribute with the corresponding key.
 getAttribute :: String -> Content i -> String
-getAttribute attr (CElem content undefined) =
-    let matchingAttrs = filter (\x -> getAttrName x == attr) $ getAttrs content
-    in if null matchingAttrs
-       then ""
-       else getAttrVal $ head matchingAttrs
+getAttribute attr (CElem content undefined)
+    | null matchingAttrs = ""
+    | otherwise = getAttrVal $ head matchingAttrs
+    where matchingAttrs = filter (\x -> getAttrName x == attr) $
+                                 getAttrs content
 getAttribute _ _ = ""
 
 -- | Parses a transform String into a tuple of Float.
 parseTransform :: String -> (Float, Float)
 parseTransform transform =
-    do let parsedTransform = splitOn "," $ drop 10 transform
-           xPos = read $ parsedTransform!!0 :: Float
-           yPos = read $ init $ parsedTransform!!1 :: Float
-       (xPos, yPos)
+    let parsedTransform = splitOn "," $ drop 10 transform
+        xPos = read $ parsedTransform!!0 :: Float
+        yPos = read $ init $ parsedTransform!!1 :: Float in
+    (xPos, yPos)
 
 -- | Parses a path's `d` attribute.
 parsePathD :: String -> [(Float, Float)]
-parsePathD d =
-    if head d == 'm'
-    then foldCoordsRel $ filter (\x -> length x > 1) $ map (splitOn ",") $ splitOn " " d
-    else processAbsCoords $ filter (\x -> length x > 1) $ map (splitOn ",") $ splitOn " " d
+parsePathD d
+    | head d == 'm' = foldCoordsRel $ filter lengthMoreThanOne $ map (splitOn ",") $ splitOn " " d
+    | otherwise =  processAbsCoords $ filter lengthMoreThanOne $ map (splitOn ",") $ splitOn " " d
+    where lengthMoreThanOne = (\x -> length x > 1)
```

Comment:
You don't need the parentheses here. Also, put the duplicated argument in the where.